src/templates/html/mapAndLabel/MapAndLabelMap.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "react";
+
+const osCopyright = `Basemap subject to Crown copyright and database rights ${new Date().getFullYear()} OS AC0000812160`;
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace JSX {
+    interface IntrinsicElements {
+      "my-map": MapProps;
+    }
+
+    interface MapProps {
+      areaUnit?: "m2" | "ha";
+      ariaLabelOlFixedOverlay?: string;
+      basemap?: string;
+      clickFeatures?: boolean;
+      clipGeojsonData?: string;
+      collapseAttributions?: boolean;
+      disableVectorTiles?: boolean;
+      drawColor?: string;
+      drawFillColor?: string;
+      drawGeojsonData?: string;
+      drawGeojsonDataBuffer?: number;
+      drawGeojsonDataCopyright?: string;
+      drawMany?: boolean;
+      drawMode?: boolean;
+      drawPointColor?: string;
+      drawPointer?: "crosshair" | "dot";
+      drawType?: "Polygon" | "Point";
+      featureBorderNone?: boolean;
+      featureBuffer?: number;
+      featureColor?: string;
+      featureFill?: string;
+      geojsonBuffer?: number;
+      geojsonColor?: string;
+      geojsonData?: string;
+      geojsonDataCopyright?: string;
+      geojsonFill?: boolean;
+      showGeojsonDataLabels?: boolean;
+      hideResetControl?: boolean;
+      id?: string;
+      latitude?: number;
+      longitude?: number;
+      markerColor?: string;
+      markerImage?: "circle" | "pin";
+      markerLatitude?: number;
+      markerLongitude?: number;
+      maxZoom?: number;
+      minZoom?: number;
+      osApiKey?: string;
+      osCopyright?: string;
+      osProxyEndpoint?: string;
+      projection?: "EPSG:4326" | "EPSG:3857" | "EPSG:27700";
+      resetControlImage?: "unicode" | "trash";
+      showCentreMarker?: boolean;
+      showFeaturesAtPoint?: string;
+      showGeojsonDataMarkers?: boolean;
+      showNorthArrow?: boolean;
+      showPrint?: boolean;
+      showScale?: boolean;
+      staticMode?: boolean;
+      style?: any;
+      useScalebarStyle?: boolean;
+      zoom?: number;
+    }
+  }
+}
+
+export default function MapAndLabelMap(props: {
+  features: object;
+  clipGeojsonData: GeoJSON.Feature;
+  drawColor: string;
+  osApiKey?: string;
+}) {
+  return (
+    <>
+      <my-map
+        id="map-and-label-map"
+        showNorthArrow={true}
+        hideResetControl={true}
+        showScale={true}
+        useScalebarStyle={true}
+        showPrint={true}
+        collapseAttributions={true}
+        basemap={props.osApiKey ? `OSVectorTile` : `OSM`}
+        osApiKey={props.osApiKey || ""}
```

Comment:
In local testing, I couldn't manage to get `osProxyEndpoint` working (I guess because of CORS?) so I was passing the OS API key directly. 

Any better ideas here? When generating an HTML _document_, is there any alternative to bundling the key directly? Should we store it as an env var in planx-core and read in that way, or pass env var from planx-new?

Reply:
Now solved via https://github.com/theopensystemslab/planx-new/pull/5208 - thanks @DafyddLlyr !

Going to pick up switch to `osProxy` in a follow-up PR :heavy_plus_sign:

Reply:
Now updated to use `osProxyEndpoint` instead ! Still pointing at `api.editor.planx.dev` because https://github.com/theopensystemslab/planx-new/pull/5208 isn't on prod yet - we can come back and adjust this later !